test/unittests/test_extend_tt.cpp

```diff
@@ -0,0 +1,140 @@
+#include "core/truthTable/truth_table.hpp"
+
+#include "gtest/gtest.h"
+
+using namespace dd::literals;
+using namespace syrec;
+
+class TruthTableExtend: public testing::TestWithParam<std::string> {
+protected:
+    dd::QubitCount                 nqubits = 3U;
+    TruthTable                     tt{};
+    std::unique_ptr<dd::Package<>> dd;
+
+    TruthTable::Cube::Value emptyVal;
+
+    TruthTable::Cube c00{};
+    TruthTable::Cube c01{};
+    TruthTable::Cube c10{};
+    TruthTable::Cube c11{};
+
+    TruthTable::Cube c000{};
+    TruthTable::Cube c001{};
+    TruthTable::Cube c010{};
+    TruthTable::Cube c011{};
+    TruthTable::Cube c100{};
+    TruthTable::Cube c101{};
+    TruthTable::Cube c110{};
+    TruthTable::Cube c111{};
+
+    TruthTable::Cube c_11{};
+    TruthTable::Cube c1_0{};
+
+    void SetUp() override {
+        dd = std::make_unique<dd::Package<>>(nqubits);
+
+        c00 = TruthTable::Cube::fromInteger(0b00U, 2U);
+        c01 = TruthTable::Cube::fromInteger(0b01U, 2U);
+        c10 = TruthTable::Cube::fromInteger(0b10U, 2U);
+        c11 = TruthTable::Cube::fromInteger(0b11U, 2U);
+
+        c000 = TruthTable::Cube::fromInteger(0b000U, 3U);
+        c001 = TruthTable::Cube::fromInteger(0b001U, 3U);
+        c010 = TruthTable::Cube::fromInteger(0b010U, 3U);
+        c011 = TruthTable::Cube::fromInteger(0b011U, 3U);
+        c100 = TruthTable::Cube::fromInteger(0b100U, 3U);
+        c101 = TruthTable::Cube::fromInteger(0b101U, 3U);
+        c110 = TruthTable::Cube::fromInteger(0b110U, 3U);
+        c111 = TruthTable::Cube::fromInteger(0b111U, 3U);
```

Comment:
now that you implemented the PLA parser, this should be refactored appropriately. See the comment on the DD synthesis test file for more information.

Reply:
have removed the most of the cubes, however, I have kept a few for checking whether particular cube instances are added or not.